### ISSUE TITLE
using lasagne.layers.BatchNormLayer

### ISSUE THREAD

**Issue by kungfujam** [dnouri/nolearn#282]
I'm struggling to include a layer with batch normalisation because it must be inserted "between a linear transformation (such as a DenseLayer, or Conv2DLayer) and its nonlinearity".

Is there a way to easily add a batch normalisation layer within the layers argument of the NeuralNet class?

**Comment by kungfujam** [comment]
Scratch this. Was being stupid. This runs fine:

```
from lasagne import layers
from lasagne.updates import adam
from lasagne.nonlinearities import softmax, rectify
from nolearn.lasagne import NeuralNet
from sklearn.datasets import make_classification
X, y = make_classification(1000, 200)
X_conv = X.reshape(-1,1,200).astype('float32')
y = y.astype('int32')
conv1 = NeuralNet(
    layers=[
        (layers.InputLayer,
         {'name': 'input', 'shape': (None, 1, X_conv.shape[2])}),
        (layers.BatchNormLayer, {'name': 'batchnorm1'}),
        (layers.Conv1DLayer,
         {'name': 'conv1', 'incoming': 'batchnorm1', 'num_filters': 16, 'filter_size': 4}),
        (layers.FlattenLayer, {'name': 'conv1_flat', 'outdim': 2}),
        (layers.Conv1DLayer,
         {'name': 'conv2', 'incoming': 'conv1', 'num_filters': 32, 'filter_size': 16}),
        (layers.FlattenLayer, {'name': 'conv2_flat', 'outdim': 2}),
        (layers.Conv1DLayer,
         {'name': 'conv3', 'incoming': 'conv1', 'num_filters': 64, 'filter_size': 64}),
        (layers.FlattenLayer, {'name': 'conv3_flat', 'outdim': 2}),
        (layers.ConcatLayer,
         {'incomings': ['conv2_flat', 'conv3_flat']}),
        (layers.DenseLayer,
         {'name': 'hidden'}),
        (layers.DenseLayer,
         {'name': 'output'}),
        ],
    hidden_num_units=100, hidden_nonlinearity=rectify, 
    output_nonlinearity=softmax,
    output_num_units=1,
    update=adam,
    verbose=1
    )
conv1.fit(X_conv, y, epochs=100)
```
